### DISCUSSION TITLE
Logging out from Web UI still raises Airflow 405 error

### DISCUSSION THREAD

**Discussion by punx120** [apache/airflow#45360] [General]
### Apache Airflow version

2.10.4

### What happened?

I still have the same error as described in #40470 even though i'm running 2.10.4 with apache-airflow-providers-fab             1.4.0. It's probably linked to some package - but i did ran
`pip install "apache-airflow==2.10.4" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.10.4/constraints-3.8.txt"`

### What you think should happen instead?

logout process needs to complete correctly

### How to reproduce

connected as admin and try to logout

### Operating System

Fedora 8.3

### Versions of Apache Airflow Providers

apache-airflow-providers-fab==1.4.0

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Spaarsh** [comment]
Hey! I would like to work on this issue!

**Comment by Spaarsh** [comment]
(I have already posted a comment under the referenced issue and am posting the same comment here as well.)

It took me a while to triage this issue. I went through the main-branch code as well as the last released source code of airflow (2.10.4). I think I may have a way of solving this. Please correct me if I am wrong!

### Triaging
I thoroughly went through the entirety of the auth-related code (to the best of my ability). There is no wrong method being called. But since the FAB requires us to not allow any GET requests at the /logout API endpoint, the HTML can't be rendered when a user manually enters the URL in the browser point.

Before moving further though, I wanted to ensure that the /logout endpoint does work for POST requests. There is one instance of a JS code sending a POST request to the /logout endpoint via the ```no_roles_permissions.html```, where there is a logout button. In order to test this, I created a user with no roles using the command:
```
airflow users create \
    --username no_roles_user \
    --firstname No \
    --lastname Roles \
    --email no_roles_user@example.com \
    --password your_password \
    --role Public
```

When I then went to the /home page, the expected page showed up:
![image](https://github.com/user-attachments/assets/3593d7d9-3c63-4475-a674-afad514607c4)

When I clicked on the "logout" button, I was successfully able to log out, indicating no fault at the endpoint itself.

### Solution
Hence my solution is as follows:
We can create a new endpoint such as ```/logout_page``` which renders an HTML which has a logout button. The logout button has a JS event handler that sends the POST request to the /logout endpoint, resulting in successful logout using GUI.

This way, the GET request doesn't happen on our /logout endpoint itself (thus not violating any FAB requirements) while also enabling a GUI-based logout action.

If this is the correct approach, I am willing to open a PR.

**Comment by shahar1** [comment]
@Spaarsh well analyzed, solution seems solid.
Feel free to tag me in your PR for review

**Comment by Spaarsh** [comment]
@shahar1 Sure! I had a question though. Should I add the logout button on some existing html page? Or provide a hyperlink on some existing page?

**Comment by shahar1** [comment]
> @shahar1 Sure! I had a question though. Should I add the logout button on some existing html page? Or provide a hyperlink on some existing page?

Just try to make it as much as trivial to the user, as logging out should be a simple action. If you encounter any limitations, we could discuss them in the PR.

**Comment by Spaarsh** [comment]
@punx120 could you please tell us the steps to recreate this error? That will greatly help in fixing it. Thanks!

**Comment by punx120** [comment]
> @punx120 could you please tell us the steps to recreate this error? That will greatly help in fixing it. Thanks!
![Screenshot From 2025-01-02 10-01-50](https://github.com/user-attachments/assets/26814a85-6c61-48ca-902e-2808e973ca21)

I simply click on the logout link.

**Comment by punx120** [comment]
On a fresh install, it works - so it must be a config and/or package conflict - will try to dig into it

**Comment by Spaarsh** [comment]
Next time the error occurs, inspect the logout button and check the corresponding action that is being taken on clicking it. From what I have discerned, the error should only surface if a GET request is sent at the ```/logout/``` endpoint. Check in the webserver logs as well and check the request being sent. If it is GET, then we may have a lead.

**Comment by potiuk** [comment]
And take a look also at the providers reported by webserver (in the webserver menu). It could be that your webserver had - for whatever reason - different provider installed - or maybe even different airflow+provider installed when you run it. And it this case the error is on the side of provider installed in the webserver.

**Comment by potiuk** [comment]
I convert it to a discussion as well - as it's something differnt - the issue seems to be solved but we might still discuss what's the problem you encountered with packages.

**Comment by punx120** [comment]
Ok i found the issue - it's because I have a custom `AuthOAuthView` that override `logout` method. Adding `methods=["POST"]` fixes the issue (basically same as #43592) - do we even need a _custom_ OAuthView?

**Comment by pykenny** [comment]
**(Updated)** We were getting the same issue using `airflow.providers.fab.auth_manager.api.auth.backend.basic_auth`, and it turned out our `fab` provider's still on 1.1, and upgrading `fab` provider (in our case somewhere 1.4+) works.
